core/src/main/java/org/elasticsearch/common/blobstore/fs/FsBlobContainer.java

```diff
@@ -85,7 +85,10 @@ public FsBlobContainer(FsBlobStore blobStore, BlobPath blobPath, Path path) {
     @Override
     public void deleteBlob(String blobName) throws IOException {
         Path blobPath = path.resolve(blobName);
-        Files.deleteIfExists(blobPath);
+        if (!Files.deleteIfExists(blobPath)) {
+            // blobPath does not exist
+            throw new IOException("File [" + blobPath.toString() + "] does not exist");
+        }
```

Comment:
@gfyoung @rmuir's suggestion here is correct.  If you can fix that, I'll merge the PR; sorry I was a bit delayed today in responding.